jenkins/igt/Jenkinsfile

```diff
@@ -0,0 +1,69 @@
+pipeline {
+  options { timestamps() }
+  agent { dockerfile { dir 'jenkins/igt/' } }
+
+  stages {
+    stage("Setup") {
+      steps {
+        script {
+          env.PIPELINE_VERSION = VersionNumber(
+             versionNumberString: '${BUILD_DATE_FORMATTED,"yyyyMMdd"}.${BUILDS_TODAY_Z}')
+        }
+        sh 'mkdir ${PIPELINE_VERSION}'
+      }
+    }
+
+    stage("Build libdrm") {
+      steps {
+        dir("${PIPELINE_VERSION}") {
+          sh 'mkdir libdrm'
+          dir("libdrm") {
+            git url: 'git://anongit.freedesktop.org/mesa/drm'
+            sh 'PKG_CONFIG_PATH=/usr/lib/arm-linux-gnueabihf/pkgconfig/ sh autogen.sh --host=arm-linux-gnueabihf --enable-intel --prefix=$(pwd)/../libdrm-build/'
+            sh 'make -j4 V=1'
+            sh 'make -j4 install V=1'
+            sh 'tar cvfz ../libdrm-build.tar.gz ../libdrm-build/'
+          }
+        }
+      }
+    }
+
+    stage("Build igt-gpu-tools") {
+      steps {
+        dir("${PIPELINE_VERSION}") {
+          sh 'mkdir igt-gpu-tools'
+          dir("igt-gpu-tools") {
+            git url: 'git://anongit.freedesktop.org/drm/igt-gpu-tools'
+            sh 'PKG_CONFIG_PATH=/usr/lib/arm-linux-gnueabihf/pkgconfig/:$(pwd)/../libdrm-build/lib/pkgconfig sh autogen.sh --host=arm-linux-gnueabihf'
+            sh 'make -j4 V=1'
+            sh 'tar cvfz ../igt-build.tar.gz tests/core_auth tests/core_get_client_auth tests/core_getclient tests/core_getstats tests/core_getversion tests/core_prop_blob tests/core_setmaster_vs_auth tests/drm_read tests/kms_addfb_basic tests/kms_atomic tests/kms_flip_event_leak tests/kms_setmode tests/kms_vblank tests/kms_frontbuffer_tracking tests/kms_flip tests/testdisplay'
+            sh ' rm -rf ../libdrm-build'
+            sh ' rm -rf ../libdrm'
+            deleteDir()
```

Comment:
It would probably be better to do this kind of clean-up at the end of the job, there's a syntax for this:
https://jenkins.io/doc/book/pipeline/syntax/#post

One advantage being that it will get run even if something failed and caused the job to abort.  I also find it a bit neater.